### PR TITLE
libc/stdio: Enable LIBC_FLOATINGPOINT by default if ARCH_FPU is enabled

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
and DEFAULT_SMALL is false

## Impact
avoid the wrong configuration for the newbie

## Testing
Pass CI

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Comment by pkarashchenko** [comment]
Looks like at least `boards/arm/stm32f7/nucleo-144/configs/f746-pysim/defconfig` should be modified as well.

File: libs/libc/stdio/Kconfig [hunk 1]
```diff
@@ -53,7 +53,7 @@ config NUNGET_CHARS
 
 config LIBC_FLOATINGPOINT
 	bool "Enable floating point in printf"
-	default n
+	default !DEFAULT_SMALL && ARCH_FPU
```
Comment (pkarashchenko) [inline libs/libc/stdio/Kconfig:56]: Why we rely on `ARCH_FPU`? The `LIBC_FLOATINGPOINT` does not have anything about hard vs soft FPU. Or for archs with `ARCH_FPU` the code size increase is low?

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
`boards/risc-v/mpfs/m100pfsevp/configs/nsh/defconfig` needs to be updated

**Comment by pkarashchenko** [comment]
@hartmannathan could you please review?

**Comment by hartmannathan** [comment]
> @hartmannathan could you please review?

Sure, i'll review soon. Thanks

**Pull request merged**